here are node tables that are not `equals( )` and yet have no detectable differences

This tree sequence:  [ts1.zip](https://github.com/user-attachments/files/21539566/ts1.zip) and this one: [ts2.zip](https://github.com/user-attachments/files/21539568/ts2.zip) (note, these are not tszipped despite the suffix)  have node tables that if you do
`tsa.tables.nodes.assert_equals(tb.tables.nodes)` then you get:
```
Traceback (most recent call last):
  File "<python-input-28>", line 1, in <module>
    ta.assert_equals(tb)
    ~~~~~~~~~~~~~~~~^^^^
  File "/home/peter/projects/tskit-dev/tsvenv/lib/python3.13/site-packages/tskit/tables.py", line 465, in assert_equals
    raise AssertionError(
    ^^^^^^^^^^^^^^^^^^^^^
    ...<2 lines>...
    )  # pragma: no cover
    ^
AssertionError: Tables differ in an undetected way - this is a bug, please report an issue on gitub
```
Indeed, examining their `flags`, `population`, `individual`, `time`, `metadata`, and `metadata_offset` vectors and their `metadata_schema` individually says they're all identical. The way [this error comes up](https://github.com/tskit-dev/tskit/blob/c23aafcfc3bd49bc41955a3f509f50d0fbf4b840/python/tskit/tables.py#L2412)  is if the lower-level tree sequence `equals` method returns `False` but then we can't find the difference in python.

The problem seems to be in metadata, since doing
```
>>> tsa.tables.assert_equals(tsb.tables, ignore_metadata=True)
```
does not produce this error.

--- a/python/CHANGELOG.rst
+++ b/python/CHANGELOG.rst
@@ -19,12 +19,15 @@
 - Add ``TreeSequence.mutations_edge`` which returns the edge ID for each mutation's
   edge. (:user:`benjeffery`, :pr:`3226`, :issue:`3189`)
 
-
 **Bugfixes**
 
 - Fix bug in ``TreeSequence.pair_coalescence_counts`` when ``span_normalise=True``
   and a window breakpoint falls within an internal missing interval.
   (:user:`nspope`, :pr:`3176`, :issue:`3175`)
+
+- Fix metadata schemas that are equal but have different byte representations not being equal
+  when using ``TableCollection.assert_equals`` and ``Table.assert_equals``.
+  (:user:`benjeffery`, :pr:`3246`, :issue:`3244`)
 
 **Breaking changes** 
 

--- a/python/tests/test_tables.py
+++ b/python/tests/test_tables.py
@@ -1401,6 +1401,21 @@ class AssertEqualsMixin:
                 table_5row.assert_equals(table2)
             table_5row.assert_equals(table2, ignore_metadata=True)
 
+    def test_degenerate_metadata_schema(self, table_5row):
+        if hasattr(table_5row, "metadata_schema"):
+            table_5row.clear()
+            table1 = table_5row.copy()
+            table2 = table_5row.copy()
+            table1.ll_table.metadata_schema = (
+                '{"codec": "json", "properties": '
+                '{"A": {"type": "integer"}, "B": {"type": "number"}}}'
+            )
+            table2.ll_table.metadata_schema = (
+                '{"codec": "json", "properties": '
+                '{"B": {"type": "number"}, "A": {"type": "integer"}}}'
+            )
+            table1.assert_equals(table2)
+
     def test_row_changes(self, table_5row, test_rows):
         for column_name in test_rows[0].keys():
             table2 = self.table_class()
@@ -4216,6 +4231,17 @@ class TestTableCollectionAssertEquals:
         ):
             t1.assert_equals(t2)
         t1.assert_equals(t2, ignore_reference_sequence=True)
+
+    def test_degenerate_metadata_schema(self, t1, t2):
+        t1._ll_object.metadata_schema = (
+            '{"codec": "json", "properties": '
+            '{"A": {"type": "integer"}, "B": {"type": "number"}}}'
+        )
+        t2._ll_object.metadata_schema = (
+            '{"codec": "json", "properties": '
+            '{"B": {"type": "number"}, "A": {"type": "integer"}}}'
+        )
+        t1.assert_equals(t2)
 
 
 class TestTableCollectionMethodSignatures:

--- a/python/tskit/tables.py
+++ b/python/tskit/tables.py
@@ -462,9 +462,18 @@ class BaseTable:
                 f"other={other.num_rows}"
             )
 
+        # We can reach this point if the metadata schemas byte representations
+        # differ when the decoded schema is the same
+        if (
+            not ignore_metadata
+            and self.ll_table.metadata_schema != other.ll_table.metadata_schema
+            and self.metadata_schema == other.metadata_schema
+        ):
+            return
+
         raise AssertionError(
             "Tables differ in an undetected way - "
-            "this is a bug, please report an issue on gitub"
+            "this is a bug, please report an issue on github"
         )  # pragma: no cover
 
     def __eq__(self, other):
@@ -2411,7 +2420,7 @@ class ProvenanceTable(BaseTable):
 
         raise AssertionError(
             "Tables differ in an undetected way - "
-            "this is a bug, please report an issue on gitub"
+            "this is a bug, please report an issue on github"
         )  # pragma: no cover
 
     def add_row(self, record, timestamp=None):
@@ -3277,9 +3286,18 @@ class TableCollection(metadata.MetadataProvider):
                 other.provenances, ignore_timestamps=ignore_timestamps
             )
 
+        # We can reach this point if the metadata schemas byte representations
+        # differ when the decoded schema is the same
+        if (
+            not ignore_metadata
+            and self._ll_object.metadata_schema != other._ll_object.metadata_schema
+            and self.metadata_schema == other.metadata_schema
+        ):
+            return
+
         raise AssertionError(
             "TableCollections differ in an undetected way - "
-            "this is a bug, please report an issue on gitub"
+            "this is a bug, please report an issue on github"
         )  # pragma: no cover
 
     def __eq__(self, other):

